### PR TITLE
Update release notes for 2.8.2 release

### PR DESCRIPTION
**Pull request by niloc132**
This change is [Reviewable](https://reviewable.io/reviews/gwtproject/gwt-site/260)

### CODE REVIEW

File: src/main/markdown/release-notes.md [hunk 1]
```diff
@@ -61,6 +62,39 @@ The GWT Release Notes
...
+standards.
+
+### Bug fixes
+- LookupMethodCreator creates too large method
+- NativeRegExp should use iframe instance, fixing Edge JIT bug
+- JsProperty getter/setter sometimes were reported as incompatible
+- Instantiating native JsTypes from JSNI results in InternalCompilerException
+- Remove the SUBSIZED characteristic from filtered streams
+- Internal compiler exception when using native JsType varargs in a JsMethod
+- Regression in String.toLowerCase and toUpperCase
```
Comment (tbroyer) [inline src/main/markdown/release-notes.md]: Wrt locale-sensitivity, particularly for Turkish.

File: src/main/markdown/release-notes.md [hunk 2]
```diff
@@ -61,6 +62,39 @@ The GWT Release Notes
...
+- Missing bounds check in String.charAt
+- Fix AIOOBE when compiling method references involving varargs.
+- Apply HtmlUnit workaround ensuring that window.onerror is called correctly
+
+### Miscellanous
+- Migrated lang/jre emulation JSNI to JsInterop to share with J2CL
+- Added ErrorProne to gwt builds
+- Improved compliance with CSP
+- Uncaught Exception Handler now can receive window.onerror
```
Comment (tbroyer) [inline src/main/markdown/release-notes.md]: IIRC, Daniel asked that we mark this as a "potentially breaking" change; how about moving this to the highlights section? (and it's not a "can" but "will, by default (use XXX to revert to previous behavior)"

File: src/main/markdown/release-notes.md [hunk 3]
```diff
@@ -61,6 +62,39 @@ The GWT Release Notes
...
+- Added emulation for java.io.Externalizable
+- Added emulation for java.lang.reflect.Array
+- JSO.equals/hashcode now can delegate to the JS object
```
Comment (tbroyer) [inline src/main/markdown/release-notes.md]: Will delegate if the JS object has methods with those names.

File: src/main/markdown/release-notes.md [hunk 4]
```diff
@@ -61,6 +62,39 @@ The GWT Release Notes
 * * *
 
 <a id="Release_Notes_Current"></a>
+## <a id="Release_Notes_2_8_2"></a> Release Notes for 2.8.2
+
+### Highlights
+- Supports running in Java 9.
```
Comment (tbroyer) [inline src/main/markdown/release-notes.md]: Highlight that none of language or APIs from Java 9 is supported?

File: src/main/markdown/release-notes.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 The GWT Release Notes
 =====================
+* [2.8.2](#Release_Notes_2_8_2) Oct 18, 2017
```
Comment (tbroyer) [inline src/main/markdown/release-notes.md]: 18 or 19 ? 😉

**Comment by niloc132** [comment]
All great points, I'll make these changes.

**Comment by niloc132** [comment]
At @JamesXNelson's suggestion, I also further clarified what aspects of Java 9's changes are supported.

**Comment by niloc132** [comment]
Looks like I missed versions.md, will update

**Pull request merged**